### ISSUE TITLE
AUC about UADFV

### ISSUE THREAD

**Issue by mxtsecure** [erprogs/CViT#9]
Hi, I want to test CViT for detecting deepfake videos. I use:

1. **pretrained model** ：deepfake_cvit_gpu_inference_ep_50.pth Publish in Releases
2. **dataset**:：UADFV
3. **test script**：cvit_prediction.py

I got the result of AUC : **75.28%**.
I want to konw if it is **normal**? And Which **Dataset** is for the pretrained model? Which **pretrained model** is the best?

**Comment by erprogs** [comment]
Hi @mxtsecure, thanks for testing our model.

The CViT model is trained using [DFDC](https://www.kaggle.com/c/deepfake-detection-challenge/data) dataset.  In our test, we have found that model accuracy may depend on the face detection libraries (Blazeface, Face_recognition, MTCNN) you use. We got the following accuracy on UADF: 74.50%(Blazeface), 93.75%(Face_recognition), 88.16% (MTCNN). This is documented in the [paper](https://arxiv.org/abs/2102.11126) (Table 4). I can say your result is normal.

**Comment by erprogs** [comment]
**Which pretrained model is the best?** Both deepfake_cvit_gpu_inference_ep_50.pth & deepfake_cvit_gpu_ep_50.pth are the same model weights. deepfake_cvit_gpu_inference_ep_50.pth contains only state_dict, and is used for inference, and  deepfake_cvit_gpu_ep_50.pth is model snapshot, and can be used for further training. I hope this answers your question.

**Comment by JiangdongWang123** [comment]
Hi,how do I get AUC results in cvit_prediction.py? I didn't find the code to output the result of AUC?


> Hi @mxtsecure, thanks for testing our model.
> 
> The CViT model is trained using [DFDC](https://www.kaggle.com/c/deepfake-detection-challenge/data) dataset. In our test, we have found that model accuracy may depend on the face detection libraries (Blazeface, Face_recognition, MTCNN) you use. We got the following accuracy on UADF: 74.50%(Blazeface), 93.75%(Face_recognition), 88.16% (MTCNN). This is documented in the [paper](https://arxiv.org/abs/2102.11126) (Table 4). I can say your result is normal.

**Comment by erprogs** [comment]
Hello @JiangdongWang123,

To calculate the ROC curve, you need to store the labels of the "**True**" value of the images/videos you want to test. That is Fake or Real.

Assume you have an array which contains list of videos you want to test: we call this array: `y_prob`
`y_prob` array contains **0** or **1**. That is 0 if the video is REAL and 1 if the video is Fake.

After you run your prediction, you will need to save it in an array, lets say `y_pred`.
If you are running your prediction on the [DFDC](https://www.kaggle.com/competitions/deepfake-detection-challenge/data) sample datasets, you will need the `metadata.json` file. 

```
k=pd.read_json('path/metadata/metadata.json')
y_test = []
for i in y_pred:
    if k[i]['label'] == 'FAKE':
        y_test.append(1)
    else:
        y_test.append(0)
```
Now, you can compute the ROC curve.
```
# Compute ROC curve and area the curve
fpr, tpr, thresholds = roc_curve(y_test, y_prob)
roc_auc = auc(fpr, tpr)
print("Area under the ROC curve : %f" % roc_auc)

# Plot ROC curve
pl.clf()
plt.figure()
lw = 2
print('Accuracy score: %f' % accuracy_score(y_test, y_prob))
pl.plot(fpr, tpr, color='darkorange', label='ROC curve (area = %0.2f)' % roc_auc)
pl.plot([0, 1], [0, 1], color='navy', lw=lw,  linestyle='--')
pl.xlim([0.0, 1.0])
pl.ylim([0.0, 1.0])
pl.xlabel('False Positive Rate')
pl.ylabel('True Positive Rate')
pl.title('Receiver operating characteristic (ROC) Curve')
pl.legend(loc="lower right")
pl.show()
```
Hope this answers your question. Thanks.

**Comment by erprogs** [comment]
@JiangdongWang123 

**NB**: I haven't tested it on my system, but it should work.

**Comment by JiangdongWang123** [comment]
Hello!@[erprogs](https://github.com/erprogs)
Thanks for your reply, I ran it successfully.

**Comment by JiangdongWang123** [comment]
Hello! @erprogs 
Thanks for your reply, I ran it successfully.

**Comment by erprogs** [comment]
Hello @JiangdongWang123,

Great!
